src/openjd/adaptor_runtime/_entrypoint.py

```diff
@@ -295,17 +360,26 @@ def _handle_run(
     def _handle_daemon(
         self,
         adaptor: BaseAdaptor[AdaptorConfiguration],
-        parsed_args: Namespace,
+        parsed_args: _ParsedArgs,
         log_config: _LogConfig,
         integration_data: _IntegrationData,
         reentry_exe: Optional[Path] = None,
     ):
-        connection_file = parsed_args.connection_file
-        if not os.path.isabs(connection_file):
-            connection_file = os.path.abspath(connection_file)
+        # Validate args
         subcommand = parsed_args.subcommand if hasattr(parsed_args, "subcommand") else None
 
+        connection_file: Path | None = None
+        if hasattr(parsed_args, "connection_file") and parsed_args.connection_file:
+            connection_file = Path(parsed_args.connection_file)
+        if connection_file and not connection_file.is_absolute():
+            connection_file = connection_file.absolute()
+
         if subcommand == "_serve":
+            if not connection_file:
+                raise RuntimeError(
+                    "--connection file is required for the '_serve' command but was not provided."
+                )
+
```

Comment:
Is there a reason we don't read from the environment variable here too?

Reply:
Ah good point, I will add that ability.

Reply:
Actually, thinking about this further. The environment variable `OPENJD_ADAPTOR_SOCKET` points to the path for the socket that should be used for IPC, so that's different from the connection file. Currently, the BackendRunner is responsible for creating the socket and communicating that back to the frontend (via the --connection-file). We can change the backend to accept a path to try to create the socket into, however that currently does not happen.

Since the `_serve` command is intended to be an "internal" command and is hidden from CLI help text, I think we can skip this for now. Does that sound ok?